### PR TITLE
[OSD-20024] Add test case for GetRestConfigAsUser

### PR DESCRIPTION
**Pull request by Tof1973**
### What type of PR is this?

_test_

### What this PR does / Why we need it?
followup of #294

### Which Jira/Github issue(s) does this PR fix?

_Resolves #_

### Pre-checks (if applicable)

- [ x] Ran unit tests locally
- [ na] Validated the changes in a cluster
- [ na] Included documentation changes with PR

### CODE REVIEW

**Comment by Tof1973** [comment]
@samanthajayasinghe, here the PR to add test to cover changes in https://github.com/openshift/backplane-cli/pull/294#pullrequestreview-1782887545

File: cmd/ocm-backplane/login/login_test.go [hunk 1]
```diff
@@ -387,4 +394,40 @@ var _ = Describe("Login command", func() {
 		})
 
 	})
+
+	Context("check GetRestConfigAsUser", func() {
+
+		It("should fail if generated config do not contains provided username without elevationReasons",func () {
```
Comment (typeid) [inline cmd/ocm-backplane/login/login_test.go:400]: The test name seems a bit confusing, I would reverse it: it should succeed if ... as you are also testing 
 `Expect(err).To(BeNil())` below. A test that mentions "should fail if..." should check for a failure.

File: cmd/ocm-backplane/login/login_test.go [hunk 2]
```diff
@@ -387,4 +394,40 @@ var _ = Describe("Login command", func() {
...
+			username := "test-user"
+
+			config, err := GetRestConfigAsUser(backplaneConfiguration, testClusterID, username)
+			Expect(err).To(BeNil())
+			Expect(config.Impersonate.UserName).To(Equal(username))
+			Expect(len(config.Impersonate.Extra["reason"])).To(Equal(0))
+
+		})
+
+		It("should fail if generated config do not contains provided username/elevationReasons",func () {
```
Comment (typeid) [inline cmd/ocm-backplane/login/login_test.go:415]: Same as above.

**Pull request merged**